common/src/main/java/com/hortonworks/streamline/common/util/ProxyUtil.java

```diff
@@ -119,6 +142,27 @@ private ClassLoader getJarAddedClassLoader(String jarPath) throws MalformedURLEx
         return classLoader;
     }
 
+    private ClassLoader getJarsAddedClassLoader(File file) throws MalformedURLException {
+        ClassLoader classLoader = new MutableURLClassLoader(new URL[0], parentClassLoader);
+        File[] files;
+        if ((file != null) && ((files = file.listFiles(JAR_FILENAME_FILTER)) != null)) {
+            for (File jar : files) {
+                URL u = (jar.toURI().toURL());
+                ((MutableURLClassLoader) classLoader).addURL(u);
+            }
+            ClassLoader oldCl = cachedClassLoaders.putIfAbsent(file.getAbsolutePath(), classLoader);
```

Comment:
Yes, its in line with the other method that loads all the classes from a jar file. Meaning, if the same jar file is updated with newer classes on FS then it will fail to load the newer classes.